// @neynar/nodejs-sdk/build/api/apis/auth-address-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Neynar API
 * The Neynar API allows you to interact with the Farcaster protocol among other things. See the [Neynar docs](https://docs.neynar.com/reference) for more details.
 *
 * The version of the OpenAPI document: 3.100.0
 * Contact: team@neynar.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthAddressApi = exports.AuthAddressApiFactory = exports.AuthAddressApiFp = exports.AuthAddressApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * AuthAddressApi - axios parameter creator
 * @export
 */
const AuthAddressApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Fetches the status of a developer managed auth address by auth address
         * @summary Status by auth address
         * @param {string} address Ethereum address
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<RegisterSignedKeyForDeveloperManagedAuthAddress200Response>} A promise that resolves to a `RegisterSignedKeyForDeveloperManagedAuthAddress200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/lookup-developer-managed-auth-address)
         *
         */
        lookupDeveloperManagedAuthAddress: async (address, options = {}) => {
            // verify required parameter 'address' is not null or undefined
            (0, common_1.assertParamExists)('lookupDeveloperManagedAuthAddress', 'address', address);
            const localVarPath = `/v2/farcaster/auth_address/developer_managed/`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (address !== undefined) {
                localVarQueryParameter['address'] = address;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Allow apps to register an Ethereum addresses as authorized \"auth addresses\" for a user\'s Farcaster account, enabling seamless Sign-In With Farcaster (SIWF) across applications without repeated custody wallet authorizations.
         * @summary Register Signed Key
         * @param {RegisterAuthAddressDeveloperManagedSignedKeyReqBody} registerAuthAddressDeveloperManagedSignedKeyReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<RegisterSignedKeyForDeveloperManagedAuthAddress200Response>} A promise that resolves to a `RegisterSignedKeyForDeveloperManagedAuthAddress200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/register-signed-key-for-developer-managed-auth-address)
         *
         */
        registerSignedKeyForDeveloperManagedAuthAddress: async (registerAuthAddressDeveloperManagedSignedKeyReqBody, options = {}) => {
            // verify required parameter 'registerAuthAddressDeveloperManagedSignedKeyReqBody' is not null or undefined
            (0, common_1.assertParamExists)('registerSignedKeyForDeveloperManagedAuthAddress', 'registerAuthAddressDeveloperManagedSignedKeyReqBody', registerAuthAddressDeveloperManagedSignedKeyReqBody);
            const localVarPath = `/v2/farcaster/auth_address/developer_managed/signed_key/`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(registerAuthAddressDeveloperManagedSignedKeyReqBody, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.AuthAddressApiAxiosParamCreator = AuthAddressApiAxiosParamCreator;
/**
 * AuthAddressApi - functional programming interface
 * @export
 */
const AuthAddressApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.AuthAddressApiAxiosParamCreator)(configuration);
    return {
        /**
         * Fetches the status of a developer managed auth address by auth address
         * @summary Status by auth address
         * @param {string} address Ethereum address
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<RegisterSignedKeyForDeveloperManagedAuthAddress200Response>} A promise that resolves to a `RegisterSignedKeyForDeveloperManagedAuthAddress200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/lookup-developer-managed-auth-address)
         *
         */
        async lookupDeveloperManagedAuthAddress(address, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.lookupDeveloperManagedAuthAddress(address, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AuthAddressApi.lookupDeveloperManagedAuthAddress']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Allow apps to register an Ethereum addresses as authorized \"auth addresses\" for a user\'s Farcaster account, enabling seamless Sign-In With Farcaster (SIWF) across applications without repeated custody wallet authorizations.
         * @summary Register Signed Key
         * @param {RegisterAuthAddressDeveloperManagedSignedKeyReqBody} registerAuthAddressDeveloperManagedSignedKeyReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<RegisterSignedKeyForDeveloperManagedAuthAddress200Response>} A promise that resolves to a `RegisterSignedKeyForDeveloperManagedAuthAddress200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/register-signed-key-for-developer-managed-auth-address)
         *
         */
        async registerSignedKeyForDeveloperManagedAuthAddress(registerAuthAddressDeveloperManagedSignedKeyReqBody, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.registerSignedKeyForDeveloperManagedAuthAddress(registerAuthAddressDeveloperManagedSignedKeyReqBody, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['AuthAddressApi.registerSignedKeyForDeveloperManagedAuthAddress']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.AuthAddressApiFp = AuthAddressApiFp;
/**
 * AuthAddressApi - factory interface
 * @export
 */
const AuthAddressApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.AuthAddressApiFp)(configuration);
    return {
        /**
         * Fetches the status of a developer managed auth address by auth address
         * @summary Status by auth address
         * @param {AuthAddressApiLookupDeveloperManagedAuthAddressRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<RegisterSignedKeyForDeveloperManagedAuthAddress200Response>} A promise that resolves to a `RegisterSignedKeyForDeveloperManagedAuthAddress200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/lookup-developer-managed-auth-address)
         *
         */
        lookupDeveloperManagedAuthAddress(requestParameters, options) {
            return localVarFp.lookupDeveloperManagedAuthAddress(requestParameters.address, options).then((request) => request(axios, basePath));
        },
        /**
         * Allow apps to register an Ethereum addresses as authorized \"auth addresses\" for a user\'s Farcaster account, enabling seamless Sign-In With Farcaster (SIWF) across applications without repeated custody wallet authorizations.
         * @summary Register Signed Key
         * @param {AuthAddressApiRegisterSignedKeyForDeveloperManagedAuthAddressRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<RegisterSignedKeyForDeveloperManagedAuthAddress200Response>} A promise that resolves to a `RegisterSignedKeyForDeveloperManagedAuthAddress200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/register-signed-key-for-developer-managed-auth-address)
         *
         */
        registerSignedKeyForDeveloperManagedAuthAddress(requestParameters, options) {
            return localVarFp.registerSignedKeyForDeveloperManagedAuthAddress(requestParameters.registerAuthAddressDeveloperManagedSignedKeyReqBody, options).then((request) => request(axios, basePath));
        },
    };
};
exports.AuthAddressApiFactory = AuthAddressApiFactory;
/**
 * AuthAddressApi - object-oriented interface
 * @export
 * @class AuthAddressApi
 * @extends {BaseAPI}
 */
class AuthAddressApi extends base_1.BaseAPI {
    /**
     * Fetches the status of a developer managed auth address by auth address
     * @summary Status by auth address
     * @param {AuthAddressApiLookupDeveloperManagedAuthAddressRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AuthAddressApi
     * @returns {Promise<RegisterSignedKeyForDeveloperManagedAuthAddress200Response>} A promise that resolves to a `RegisterSignedKeyForDeveloperManagedAuthAddress200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/lookup-developer-managed-auth-address)
     *
     */
    lookupDeveloperManagedAuthAddress(requestParameters, options) {
        return (0, exports.AuthAddressApiFp)(this.configuration).lookupDeveloperManagedAuthAddress(requestParameters.address, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Allow apps to register an Ethereum addresses as authorized \"auth addresses\" for a user\'s Farcaster account, enabling seamless Sign-In With Farcaster (SIWF) across applications without repeated custody wallet authorizations.
     * @summary Register Signed Key
     * @param {AuthAddressApiRegisterSignedKeyForDeveloperManagedAuthAddressRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof AuthAddressApi
     * @returns {Promise<RegisterSignedKeyForDeveloperManagedAuthAddress200Response>} A promise that resolves to a `RegisterSignedKeyForDeveloperManagedAuthAddress200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/register-signed-key-for-developer-managed-auth-address)
     *
     */
    registerSignedKeyForDeveloperManagedAuthAddress(requestParameters, options) {
        return (0, exports.AuthAddressApiFp)(this.configuration).registerSignedKeyForDeveloperManagedAuthAddress(requestParameters.registerAuthAddressDeveloperManagedSignedKeyReqBody, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.AuthAddressApi = AuthAddressApi;
