// @neynar/nodejs-sdk/build/api/apis/login-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Neynar API
 * The Neynar API allows you to interact with the Farcaster protocol among other things. See the [Neynar docs](https://docs.neynar.com/reference) for more details.
 *
 * The version of the OpenAPI document: 3.100.0
 * Contact: team@neynar.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoginApi = exports.LoginApiFactory = exports.LoginApiFp = exports.LoginApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * LoginApi - axios parameter creator
 * @export
 */
const LoginApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Nonce to sign a message
         * @summary Fetch nonce
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<NonceResponse>} A promise that resolves to a `NonceResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-nonce)
         *
         */
        fetchNonce: async (options = {}) => {
            const localVarPath = `/v2/farcaster/login/nonce/`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.LoginApiAxiosParamCreator = LoginApiAxiosParamCreator;
/**
 * LoginApi - functional programming interface
 * @export
 */
const LoginApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.LoginApiAxiosParamCreator)(configuration);
    return {
        /**
         * Nonce to sign a message
         * @summary Fetch nonce
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<NonceResponse>} A promise that resolves to a `NonceResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-nonce)
         *
         */
        async fetchNonce(options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.fetchNonce(options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['LoginApi.fetchNonce']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.LoginApiFp = LoginApiFp;
/**
 * LoginApi - factory interface
 * @export
 */
const LoginApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.LoginApiFp)(configuration);
    return {
        /**
         * Nonce to sign a message
         * @summary Fetch nonce
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<NonceResponse>} A promise that resolves to a `NonceResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-nonce)
         *
         */
        fetchNonce(options) {
            return localVarFp.fetchNonce(options).then((request) => request(axios, basePath));
        },
    };
};
exports.LoginApiFactory = LoginApiFactory;
/**
 * LoginApi - object-oriented interface
 * @export
 * @class LoginApi
 * @extends {BaseAPI}
 */
class LoginApi extends base_1.BaseAPI {
    /**
     * Nonce to sign a message
     * @summary Fetch nonce
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof LoginApi
     * @returns {Promise<NonceResponse>} A promise that resolves to a `NonceResponse` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-nonce)
     *
     */
    fetchNonce(options) {
        return (0, exports.LoginApiFp)(this.configuration).fetchNonce(options).then((request) => request(this.axios, this.basePath));
    }
}
exports.LoginApi = LoginApi;
