// @neynar/nodejs-sdk/build/api/apis/topic-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Neynar API
 * The Neynar API allows you to interact with the Farcaster protocol among other things. See the [Neynar docs](https://docs.neynar.com/reference) for more details.
 *
 * The version of the OpenAPI document: 3.100.0
 * Contact: team@neynar.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopicApi = exports.TopicApiFactory = exports.TopicApiFp = exports.TopicApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * TopicApi - axios parameter creator
 * @export
 */
const TopicApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Returns a list of trending topics for casts.
         * @summary Fetch trending topics
         * @param {number} [limit] Number of topics to fetch.  (Default: 10, Maximum: 25)
         * @param {string} [cursor] Pagination cursor.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<TrendingTopicsResponse>} A promise that resolves to a `TrendingTopicsResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-trending-topics)
         *
         */
        listTrendingTopics: async (limit, cursor, options = {}) => {
            const localVarPath = `/v2/farcaster/topic/trending/`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (cursor !== undefined) {
                localVarQueryParameter['cursor'] = cursor;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.TopicApiAxiosParamCreator = TopicApiAxiosParamCreator;
/**
 * TopicApi - functional programming interface
 * @export
 */
const TopicApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.TopicApiAxiosParamCreator)(configuration);
    return {
        /**
         * Returns a list of trending topics for casts.
         * @summary Fetch trending topics
         * @param {number} [limit] Number of topics to fetch.  (Default: 10, Maximum: 25)
         * @param {string} [cursor] Pagination cursor.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<TrendingTopicsResponse>} A promise that resolves to a `TrendingTopicsResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-trending-topics)
         *
         */
        async listTrendingTopics(limit, cursor, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.listTrendingTopics(limit, cursor, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['TopicApi.listTrendingTopics']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.TopicApiFp = TopicApiFp;
/**
 * TopicApi - factory interface
 * @export
 */
const TopicApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.TopicApiFp)(configuration);
    return {
        /**
         * Returns a list of trending topics for casts.
         * @summary Fetch trending topics
         * @param {TopicApiListTrendingTopicsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<TrendingTopicsResponse>} A promise that resolves to a `TrendingTopicsResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-trending-topics)
         *
         */
        listTrendingTopics(requestParameters = {}, options) {
            return localVarFp.listTrendingTopics(requestParameters.limit, requestParameters.cursor, options).then((request) => request(axios, basePath));
        },
    };
};
exports.TopicApiFactory = TopicApiFactory;
/**
 * TopicApi - object-oriented interface
 * @export
 * @class TopicApi
 * @extends {BaseAPI}
 */
class TopicApi extends base_1.BaseAPI {
    /**
     * Returns a list of trending topics for casts.
     * @summary Fetch trending topics
     * @param {TopicApiListTrendingTopicsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof TopicApi
     * @returns {Promise<TrendingTopicsResponse>} A promise that resolves to a `TrendingTopicsResponse` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-trending-topics)
     *
     */
    listTrendingTopics(requestParameters = {}, options) {
        return (0, exports.TopicApiFp)(this.configuration).listTrendingTopics(requestParameters.limit, requestParameters.cursor, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.TopicApi = TopicApi;
