// @neynar/nodejs-sdk/build/hub-api/apis/fids-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Neynar docs](https://docs.neynar.com/reference) for more details.
 *
 * The version of the OpenAPI document: 2.35.0
 * Contact: team@neynar.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FidsApi = exports.FidsApiFactory = exports.FidsApiFp = exports.FidsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * FidsApi - axios parameter creator
 * @export
 */
const FidsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Fetch a list of all the FIDs.
         * @summary Fetch a list of all the FIDs
         * @param {number} shardId The shard ID to filter by
         * @param {number} [pageSize] Maximum number of messages to return in a single response
         * @param {boolean} [reverse] Reverse the sort order, returning latest messages first
         * @param {string} [pageToken] The page token returned by the previous query, to fetch the next page. If this parameter is empty, fetch the first page
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<FidsResponse>} A promise that resolves to a `FidsResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-fids)
         *
         */
        fetchFids: async (shardId, pageSize, reverse, pageToken, options = {}) => {
            // verify required parameter 'shardId' is not null or undefined
            (0, common_1.assertParamExists)('fetchFids', 'shardId', shardId);
            const localVarPath = `/v1/fids`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (shardId !== undefined) {
                localVarQueryParameter['shard_id'] = shardId;
            }
            if (pageSize !== undefined) {
                localVarQueryParameter['pageSize'] = pageSize;
            }
            if (reverse !== undefined) {
                localVarQueryParameter['reverse'] = reverse;
            }
            if (pageToken !== undefined) {
                localVarQueryParameter['pageToken'] = pageToken;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.FidsApiAxiosParamCreator = FidsApiAxiosParamCreator;
/**
 * FidsApi - functional programming interface
 * @export
 */
const FidsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.FidsApiAxiosParamCreator)(configuration);
    return {
        /**
         * Fetch a list of all the FIDs.
         * @summary Fetch a list of all the FIDs
         * @param {number} shardId The shard ID to filter by
         * @param {number} [pageSize] Maximum number of messages to return in a single response
         * @param {boolean} [reverse] Reverse the sort order, returning latest messages first
         * @param {string} [pageToken] The page token returned by the previous query, to fetch the next page. If this parameter is empty, fetch the first page
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<FidsResponse>} A promise that resolves to a `FidsResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-fids)
         *
         */
        async fetchFids(shardId, pageSize, reverse, pageToken, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.fetchFids(shardId, pageSize, reverse, pageToken, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['FidsApi.fetchFids']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.FidsApiFp = FidsApiFp;
/**
 * FidsApi - factory interface
 * @export
 */
const FidsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.FidsApiFp)(configuration);
    return {
        /**
         * Fetch a list of all the FIDs.
         * @summary Fetch a list of all the FIDs
         * @param {FidsApiFetchFidsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<FidsResponse>} A promise that resolves to a `FidsResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-fids)
         *
         */
        fetchFids(requestParameters, options) {
            return localVarFp.fetchFids(requestParameters.shardId, requestParameters.pageSize, requestParameters.reverse, requestParameters.pageToken, options).then((request) => request(axios, basePath));
        },
    };
};
exports.FidsApiFactory = FidsApiFactory;
/**
 * FidsApi - object-oriented interface
 * @export
 * @class FidsApi
 * @extends {BaseAPI}
 */
class FidsApi extends base_1.BaseAPI {
    /**
     * Fetch a list of all the FIDs.
     * @summary Fetch a list of all the FIDs
     * @param {FidsApiFetchFidsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FidsApi
     * @returns {Promise<FidsResponse>} A promise that resolves to a `FidsResponse` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-fids)
     *
     */
    fetchFids(requestParameters, options) {
        return (0, exports.FidsApiFp)(this.configuration).fetchFids(requestParameters.shardId, requestParameters.pageSize, requestParameters.reverse, requestParameters.pageToken, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.FidsApi = FidsApi;
