// @neynar/nodejs-sdk/build/hub-api/apis/storage-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Neynar docs](https://docs.neynar.com/reference) for more details.
 *
 * The version of the OpenAPI document: 2.35.0
 * Contact: team@neynar.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageApi = exports.StorageApiFactory = exports.StorageApiFp = exports.StorageApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * StorageApi - axios parameter creator
 * @export
 */
const StorageApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Fetch a user\'s storage limits.
         * @summary FID\'s limits
         * @param {number} fid
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<StorageLimitsResponse>} A promise that resolves to a `StorageLimitsResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/lookup-user-storage-limit)
         *
         */
        lookupUserStorageLimit: async (fid, options = {}) => {
            // verify required parameter 'fid' is not null or undefined
            (0, common_1.assertParamExists)('lookupUserStorageLimit', 'fid', fid);
            const localVarPath = `/v1/storageLimitsByFid`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.StorageApiAxiosParamCreator = StorageApiAxiosParamCreator;
/**
 * StorageApi - functional programming interface
 * @export
 */
const StorageApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.StorageApiAxiosParamCreator)(configuration);
    return {
        /**
         * Fetch a user\'s storage limits.
         * @summary FID\'s limits
         * @param {number} fid
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<StorageLimitsResponse>} A promise that resolves to a `StorageLimitsResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/lookup-user-storage-limit)
         *
         */
        async lookupUserStorageLimit(fid, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.lookupUserStorageLimit(fid, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StorageApi.lookupUserStorageLimit']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.StorageApiFp = StorageApiFp;
/**
 * StorageApi - factory interface
 * @export
 */
const StorageApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.StorageApiFp)(configuration);
    return {
        /**
         * Fetch a user\'s storage limits.
         * @summary FID\'s limits
         * @param {StorageApiLookupUserStorageLimitRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<StorageLimitsResponse>} A promise that resolves to a `StorageLimitsResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/lookup-user-storage-limit)
         *
         */
        lookupUserStorageLimit(requestParameters, options) {
            return localVarFp.lookupUserStorageLimit(requestParameters.fid, options).then((request) => request(axios, basePath));
        },
    };
};
exports.StorageApiFactory = StorageApiFactory;
/**
 * StorageApi - object-oriented interface
 * @export
 * @class StorageApi
 * @extends {BaseAPI}
 */
class StorageApi extends base_1.BaseAPI {
    /**
     * Fetch a user\'s storage limits.
     * @summary FID\'s limits
     * @param {StorageApiLookupUserStorageLimitRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StorageApi
     * @returns {Promise<StorageLimitsResponse>} A promise that resolves to a `StorageLimitsResponse` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/lookup-user-storage-limit)
     *
     */
    lookupUserStorageLimit(requestParameters, options) {
        return (0, exports.StorageApiFp)(this.configuration).lookupUserStorageLimit(requestParameters.fid, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.StorageApi = StorageApi;
