// @neynar/nodejs-sdk/build/hub-api/apis/usernames-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster Hub API
 * Perform basic queries of Farcaster state via the REST API of a Farcaster hub. See the [Neynar docs](https://docs.neynar.com/reference) for more details.
 *
 * The version of the OpenAPI document: 2.35.0
 * Contact: team@neynar.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsernamesApi = exports.UsernamesApiFactory = exports.UsernamesApiFp = exports.UsernamesApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * UsernamesApi - axios parameter creator
 * @export
 */
const UsernamesApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Fetch a proof for a username.
         * @summary Proof for a username
         * @param {string} name The Farcaster username or ENS address
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UserNameProof>} A promise that resolves to a `UserNameProof` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-username-proof-by-name)
         *
         */
        fetchUsernameProofByName: async (name, options = {}) => {
            // verify required parameter 'name' is not null or undefined
            (0, common_1.assertParamExists)('fetchUsernameProofByName', 'name', name);
            const localVarPath = `/v1/userNameProofByName`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Fetch proofs provided by a user.
         * @summary Proofs provided by an FID
         * @param {number} fid The FID being requested
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UsernameProofsResponse>} A promise that resolves to a `UsernameProofsResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-username-proofs-by-fid)
         *
         */
        fetchUsernameProofsByFid: async (fid, options = {}) => {
            // verify required parameter 'fid' is not null or undefined
            (0, common_1.assertParamExists)('fetchUsernameProofsByFid', 'fid', fid);
            const localVarPath = `/v1/userNameProofsByFid`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.UsernamesApiAxiosParamCreator = UsernamesApiAxiosParamCreator;
/**
 * UsernamesApi - functional programming interface
 * @export
 */
const UsernamesApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.UsernamesApiAxiosParamCreator)(configuration);
    return {
        /**
         * Fetch a proof for a username.
         * @summary Proof for a username
         * @param {string} name The Farcaster username or ENS address
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UserNameProof>} A promise that resolves to a `UserNameProof` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-username-proof-by-name)
         *
         */
        async fetchUsernameProofByName(name, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.fetchUsernameProofByName(name, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['UsernamesApi.fetchUsernameProofByName']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Fetch proofs provided by a user.
         * @summary Proofs provided by an FID
         * @param {number} fid The FID being requested
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UsernameProofsResponse>} A promise that resolves to a `UsernameProofsResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-username-proofs-by-fid)
         *
         */
        async fetchUsernameProofsByFid(fid, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.fetchUsernameProofsByFid(fid, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['UsernamesApi.fetchUsernameProofsByFid']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.UsernamesApiFp = UsernamesApiFp;
/**
 * UsernamesApi - factory interface
 * @export
 */
const UsernamesApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.UsernamesApiFp)(configuration);
    return {
        /**
         * Fetch a proof for a username.
         * @summary Proof for a username
         * @param {UsernamesApiFetchUsernameProofByNameRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UserNameProof>} A promise that resolves to a `UserNameProof` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-username-proof-by-name)
         *
         */
        fetchUsernameProofByName(requestParameters, options) {
            return localVarFp.fetchUsernameProofByName(requestParameters.name, options).then((request) => request(axios, basePath));
        },
        /**
         * Fetch proofs provided by a user.
         * @summary Proofs provided by an FID
         * @param {UsernamesApiFetchUsernameProofsByFidRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UsernameProofsResponse>} A promise that resolves to a `UsernameProofsResponse` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-username-proofs-by-fid)
         *
         */
        fetchUsernameProofsByFid(requestParameters, options) {
            return localVarFp.fetchUsernameProofsByFid(requestParameters.fid, options).then((request) => request(axios, basePath));
        },
    };
};
exports.UsernamesApiFactory = UsernamesApiFactory;
/**
 * UsernamesApi - object-oriented interface
 * @export
 * @class UsernamesApi
 * @extends {BaseAPI}
 */
class UsernamesApi extends base_1.BaseAPI {
    /**
     * Fetch a proof for a username.
     * @summary Proof for a username
     * @param {UsernamesApiFetchUsernameProofByNameRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof UsernamesApi
     * @returns {Promise<UserNameProof>} A promise that resolves to a `UserNameProof` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-username-proof-by-name)
     *
     */
    fetchUsernameProofByName(requestParameters, options) {
        return (0, exports.UsernamesApiFp)(this.configuration).fetchUsernameProofByName(requestParameters.name, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Fetch proofs provided by a user.
     * @summary Proofs provided by an FID
     * @param {UsernamesApiFetchUsernameProofsByFidRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof UsernamesApi
     * @returns {Promise<UsernameProofsResponse>} A promise that resolves to a `UsernameProofsResponse` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/fetch-username-proofs-by-fid)
     *
     */
    fetchUsernameProofsByFid(requestParameters, options) {
        return (0, exports.UsernamesApiFp)(this.configuration).fetchUsernameProofsByFid(requestParameters.fid, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.UsernamesApi = UsernamesApi;
