// @neynar/nodejs-sdk/build/api/models/app-host-event-type.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Neynar API
 * The Neynar API allows you to interact with the Farcaster protocol among other things. See the [Neynar docs](https://docs.neynar.com/reference) for more details.
 *
 * The version of the OpenAPI document: 3.100.0
 * Contact: team@neynar.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppHostEventType = void 0;
/**
 * Types of events that can occur between a user and an app host: - frame_added: User adds a mini app to their account - frame_removed: User removes a mini app from their account - notifications_enabled: User enables notifications for a mini app - notifications_disabled: User disables notifications for a mini app
 * @export
 * @enum {string}
 */
exports.AppHostEventType = {
    FrameAdded: 'frame_added',
    FrameRemoved: 'frame_removed',
    NotificationsEnabled: 'notifications_enabled',
    NotificationsDisabled: 'notifications_disabled'
};
