// @neynar/nodejs-sdk/build/api/models/notification-delivery.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Neynar API
 * The Neynar API allows you to interact with the Farcaster protocol among other things. See the [Neynar docs](https://docs.neynar.com/reference) for more details.
 *
 * The version of the OpenAPI document: 3.100.0
 * Contact: team@neynar.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationDeliveryStatusEnum = exports.NotificationDeliveryObjectEnum = void 0;
exports.NotificationDeliveryObjectEnum = {
    NotificationDelivery: 'notification_delivery'
};
exports.NotificationDeliveryStatusEnum = {
    Success: 'success',
    Failed: 'failed',
    InvalidToken: 'invalid_token',
    RateLimited: 'rate_limited',
    TokenNotFound: 'token_not_found',
    TokenDisabled: 'token_disabled',
    HttpError: 'http_error',
    InvalidTargetUrl: 'invalid_target_url'
};
