// @neynar/nodejs-sdk/build/api/models/trending-topic.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Neynar API
 * The Neynar API allows you to interact with the Farcaster protocol among other things. See the [Neynar docs](https://docs.neynar.com/reference) for more details.
 *
 * The version of the OpenAPI document: 3.100.0
 * Contact: team@neynar.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrendingTopicTopLevelTopicEnum = void 0;
exports.TrendingTopicTopLevelTopicEnum = {
    ArtsCulture: 'arts_culture',
    BusinessEntrepreneurs: 'business_entrepreneurs',
    CelebrityPopCulture: 'celebrity_pop_culture',
    DiariesDailyLife: 'diaries_daily_life',
    Family: 'family',
    FashionStyle: 'fashion_style',
    FilmTvVideo: 'film_tv_video',
    FitnessHealth: 'fitness_health',
    FoodDining: 'food_dining',
    Gaming: 'gaming',
    LearningEducational: 'learning_educational',
    Music: 'music',
    NewsSocialConcern: 'news_social_concern',
    OtherHobbies: 'other_hobbies',
    Relationships: 'relationships',
    ScienceTechnology: 'science_technology',
    Sports: 'sports',
    TravelAdventure: 'travel_adventure',
    YouthStudentLife: 'youth_student_life'
};
